### PR TITLE
MINOR: Eliminate warnings for AdminUtils

### PR DESCRIPTION
**Pull request by LoganZhuZzz**
*Minor modification, eliminate warnings for AdminUtils.*

### CODE REVIEW

File: server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java [hunk 1]
```diff
@@ -143,9 +143,9 @@ private static Map<Integer, List<Integer>> assignReplicasToBrokersRackUnaware(in
                                                                                   int fixedStartIndex,
                                                                                   int startPartitionId) {
         Map<Integer, List<Integer>> ret = new HashMap<>();
-        int startIndex = fixedStartIndex >= 0 ? fixedStartIndex : RAND.nextInt(brokerList.size());
+        int startIndex = determineIndex(fixedStartIndex, brokerList.size());
         int currentPartitionId = Math.max(0, startPartitionId);
-        int nextReplicaShift = fixedStartIndex >= 0 ? fixedStartIndex : RAND.nextInt(brokerList.size());
+        int nextReplicaShift = determineIndex(fixedStartIndex, brokerList.size());
```
Comment (gongxuanzhang) [inline server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java]: I don't think we need to refactor this function,the original code is straightforward enough

**Comment by gongxuanzhang** [comment]
BTW,I don't think you pulled the latest code.
before your commit,please run `./gradlew checkstyleMain checkstyleTest spotlessCheck` to ensure code style

**Comment by LoganZhuZzz** [comment]
> BTW,I don't think you pulled the latest code. before your commit,please run `./gradlew checkstyleMain checkstyleTest spotlessCheck` to ensure code style

*Thanks for that, I will use the latest code and ensure code style*

**Comment by gongxuanzhang** [comment]
LGTM,please take a look,thanks @chia7712

File: server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java [hunk 2]
```diff
@@ -166,7 +166,7 @@ private static Map<Integer, List<Integer>> assignReplicasToBrokersRackAware(int
                                                                                 int fixedStartIndex,
                                                                                 int startPartitionId) {
         Map<Integer, String> brokerRackMap = new HashMap<>();
-        brokerMetadatas.forEach(m -> brokerRackMap.put(m.id, m.rack.get()));
+        brokerMetadatas.forEach(m -> brokerRackMap.put(m.id, m.rack.orElse(null)));
```
Comment (chia7712) [inline server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java]: Is it expected that `rack` will be empty

**Review by chia7712 — COMMENTED** [review]
@zeq96 thanks for patch

File: server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java (hunk 2 above)
Comment (LoganZhuZzz) [reply server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java:169]: Before calling this function, the `brokerMetadatas` parameter has already been checked. If `rack` is empty, an `AdminOperationException` will be thrown. However, my IDE still issues a warning, so I made an equivalent replacement. Next time, I will ignore this situation.

Comment (LoganZhuZzz) [reply server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java]: > Is it expected that `rack` will be empty

Before calling this function, the parameter `brokerMetadatas` has already been checked. If `rack` is empty, an `AdminOperationException` will be thrown. However, my IDE still issues a warning, so I made an equivalent replacement. Next time, I will ignore this situation.

Comment (chia7712) [reply server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java]: > Before calling this function, the parameter brokerMetadatas has already been checked. If rack is empty, an AdminOperationException will be thrown. However, my IDE still issues a warning, so I made an equivalent replacement. Next time, I will ignore this situation.

How about moving the check into `assignReplicasToBrokersRackAware`? for example:
```java
brokerMetadatas.forEach(m -> brokerRackMap.put(m.id, m.rack.orElseThrow(() -> new AdminOperationException("Not all brokers have rack information for replica rack aware assignment."))));
```

Comment (LoganZhuZzz) [reply server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java]: > > Before calling this function, the parameter brokerMetadatas has already been checked. If rack is empty, an AdminOperationException will be thrown. However, my IDE still issues a warning, so I made an equivalent replacement. Next time, I will ignore this situation.
> 
> How about moving the check into `assignReplicasToBrokersRackAware`? for example:
> 
> ```java
> brokerMetadatas.forEach(m -> brokerRackMap.put(m.id, m.rack.orElseThrow(() -> new AdminOperationException("Not all brokers have rack information for replica rack aware assignment."))));
> ```

`great suggestion, I will do it`

File: server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java [hunk 3]
```diff
@@ -128,12 +128,10 @@ public static Map<Integer, List<Integer>> assignReplicasToBrokers(Collection<Bro
             throw new InvalidReplicationFactorException("Replication factor: " + replicationFactor + " larger than available brokers: " + brokerMetadatas.size() + ".");
         if (brokerMetadatas.stream().noneMatch(b -> b.rack.isPresent()))
             return assignReplicasToBrokersRackUnaware(nPartitions, replicationFactor, brokerMetadatas.stream().map(b -> b.id).collect(Collectors.toList()), fixedStartIndex,
-                startPartitionId);
+                    startPartitionId);
```
Comment (gongxuanzhang) [inline server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java]: please revert unrelated changes

File: server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java [hunk 4]
```diff
@@ -196,7 +194,7 @@ private static Map<Integer, List<Integer>> assignReplicasToBrokersRackAware(int
                     //    that do not have any replica, or
                     // 2. the broker has already assigned a replica AND there is one or more brokers that do not have replica assigned
                     if ((!racksWithReplicas.contains(rack) || racksWithReplicas.size() == numRacks)
-                        && (!brokersWithReplicas.contains(broker) || brokersWithReplicas.size() == numBrokers)) {
+                            && (!brokersWithReplicas.contains(broker) || brokersWithReplicas.size() == numBrokers)) {
```
Comment (gongxuanzhang) [inline server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java]: ditto

**Comment by gongxuanzhang** [comment]
You can turn this setting off if you are troubled by IDEA's automatic modification of blank line

image

**Comment by LoganZhuZzz** [comment]
> You can turn this setting off if you are troubled by IDEA's automatic modification of blank line
> 
> image

*Thanks for that*

File: server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java [hunk 5]
```diff
@@ -166,7 +164,7 @@ private static Map<Integer, List<Integer>> assignReplicasToBrokersRackAware(int
                                                                                 int fixedStartIndex,
                                                                                 int startPartitionId) {
         Map<Integer, String> brokerRackMap = new HashMap<>();
-        brokerMetadatas.forEach(m -> brokerRackMap.put(m.id, m.rack.get()));
+        brokerMetadatas.forEach(m -> brokerRackMap.put(m.id, m.rack.orElseThrow(() -> new AdminOperationException("Not all brokers have rack information for replica rack aware assignment."))));
```
Comment (chia7712) [inline server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java:167]: Could you please add UT for it? `AdminRackAwareTest` is a good class to have the new test.

**Review by chia7712 — COMMENTED** [review]
@LoganZhuZzz thanks for your patch. the change LGTM, but it would be nice to have a new test

File: server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java (hunk 5 above)
Comment (LoganZhuZzz) [reply server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java:167]: > Could you please add UT for it? `AdminRackAwareTest` is a good class to have the new test.

`Sure, I'm on it`

Comment (LoganZhuZzz) [reply server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java:167]: > Could you please add UT for it? `AdminRackAwareTest` is a good class to have the new test.

*I have added a new unit test for it. If there are any areas that could be improved, please provide me with some guidance and suggestions.*

**Review by chia7712 — APPROVED** [review]
LGTM
